dataone-indexer.yml.tmpl

```diff
@@ -0,0 +1,24 @@
+{{- with $base := (printf "configs/%s" (env "DE_ENV")) -}}
+{{- if tree (printf "%s/irods-amqp" $base) -}}
+amqp:
+  {{ with $v := (key (printf "%s/irods-amqp/uri" $base)) }}uri: {{ $v }}{{ end }}
+  {{- if tree (printf "%s/irods-amqp/exchange" $base) }}
+  exchange:
+    {{ with $v := (key (printf "%s/irods-amqp/exchange/name" $base)) }}name: {{ $v }}{{ end }}
+  {{- end }}
+  routing-key:
+    {{ with $v := (key (printf "%s/dataone/amqp-subscription" $base)) }}subscription: {{ $v }}{{ end }}
+    {{ with $v := (key (printf "%s/dataone/amqp-read-key" $base)) }}read: {{ $v }}{{ end }}
```

Comment:
Possibly undesirable behavior: if there's nothing in `configs/env/irods-amqp/`, then these two won't be put in even if there's values in their corresponding locations in the KV store, due to line 2 above checking only for the irods-amqp prefix. I think it's probably possible to amend line 2 somehow to care about both prefixes, or you could be lazier and put these two at something like `configs/env/irods-amqp/dataone/...` instead.

Reply:
Well, or I suppose another option would be to put them in the 'dataone' section of the config down below, to more closely match the structure you've set up in consul. Anyway.

Reply:
Eh, I'll put it in the `dataone` section.